stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/LegacyDateFormatter.kt

```diff
@@ -1,10 +1,33 @@
 package com.getstream.sdk.chat.utils
 
 import org.threeten.bp.LocalDateTime
+import org.threeten.bp.LocalTime
 import java.util.Date
 
 public abstract class LegacyDateFormatter : DateFormatter {
-    public abstract fun format(date: Date?): String
+    /**
+     * Formats the given date as a String.
+     */
+    public abstract fun formatDate(date: Date?): String
+
+    /**
+     * Formats the given time as a String.
+     *
+     * Note that only the hour, minute, and hour values contained by the parameter
+     * [Date] object are valid.
+     */
+    public abstract fun formatTime(time: Date?): String
+
     final override fun formatDate(localDateTime: LocalDateTime?): String =
-        format(localDateTime?.let(DateConverter::toDate))
+        formatDate(localDateTime?.let(DateConverter::toDate))
+    final override fun formatTime(localTime: LocalTime?): String =
+        formatTime(localTime?.let(::toTime))
+
+    private fun toTime(localTime: LocalTime): Date {
+        return Date().apply {
+            hours = localTime.hour
+            minutes = localTime.minute
+            seconds = localTime.second
+        }
+    }
```

Comment:
Added this only here and not to `DateConverter`, because it's not a great conversion, but it's what we need to do if we want to support `Date`